apps/hash-ai-worker-py/app/typesystem/_schema.py

```diff
@@ -14,21 +32,116 @@ class OntologyTypeSchema(BaseModel, ABC):
     schema_url: str = Field(..., alias="$schema")
 
 
-T = TypeVar("T", bound=BaseModel)
+T = TypeVar("T", bound=Schema)
 
 
-class OneOf(BaseModel, Generic[T]):
+class OneOf(Schema, Generic[T]):
     one_of: list[T] = Field(..., alias="oneOf")
 
+    async def create_model(
+        self,
+        *,
+        actor_id: UUID,
+    ) -> Annotated[Any, ...]:
+        types = [await value.create_model(actor_id=actor_id) for value in self.one_of]
+
+        class OneOfSchema:
+            @classmethod
+            def __get_pydantic_core_schema__(
+                cls,
+                _source_type: Any,  # noqa: ANN401
+                handler: GetCoreSchemaHandler,
+            ) -> CoreSchema:
+                return core_schema.union_schema([handler(t) for t in types])
+
+            @classmethod
+            def __get_pydantic_json_schema__(
+                cls,
+                schema: CoreSchema,
+                handler: GetJsonSchemaHandler,
+            ) -> JsonSchemaValue:
+                json_schema = handler(schema)
+                if any_of := json_schema.pop("anyOf", None):
+                    json_schema["oneOf"] = any_of
```

Comment:
What's this about?

Reply:
If multiple possible types are specified in the `oneOf`, Pydantic will put them inside of an `anyOf`. I'm still not 100% sure if we need `oneOf` in this situation or if `anyOf` is better in this situation, but this replaces the `anyOf` with an `oneOf`. For example if you have a property type with a `oneOf: [ <text data type>, <number data type> ]` this will make sure that it will generate a schema like this:
```json
{
    "oneOf": [
        { "type": "string" },
        { "type": "number" },
    ]
}
```

instead of 
```json
{
    "anyOf": [
        { "type": "string" },
        { "type": "number" },
    ]
}
```